src/modules/products/interfaces/product-response.interface.ts

```diff
@@ -0,0 +1,10 @@
+import { Product } from "./product.interface";
+
+export interface ProductResponse {
```

Comment:
Para tipar estas interfaces yo mismo he creado una interfaz genérica llamada PaginatedResponse en donde le pasas un genérico, que en este caso seria Product, quedaría algo asi:


```suggestion
import { PaginatedResponse } from "@/shared/interfaces/pagination/paginated-response.interface.ts";

export interface Product {
  productId: string;
  name: string;
  description: string;
  price: number;
  imageURLS: string[];
  stock: number;
  color: string;
  sizeName: string;
  variantOfName: string;
}

export type ProductResponse = PaginatedResponse<Product>;
```